compiler/rustc_expand/src/proc_macro_server.rs

```diff
@@ -636,17 +635,20 @@ impl server::SourceFile for Rustc<'_> {
 
 impl server::MultiSpan for Rustc<'_> {
     fn new(&mut self) -> Self::MultiSpan {
-        vec![]
+        MultiSpan::new()
     }
-    fn push(&mut self, spans: &mut Self::MultiSpan, span: Self::Span) {
-        spans.push(span)
+    fn push_primary_span(&mut self, multispan: &mut Self::MultiSpan, span: Self::Span) {
```

Comment:
What happens if user `push_primary_span` multiple times? Does it become a secondary span?

Reply:
From what I can tell, "secondary spans" are those that are labelled. The method itself is identical, all I did was rename it to match the internal naming of "primary span" [[ref](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.MultiSpan.html)].

Reply:
> They can be primary spans. In this case they are the locus of the error, and would be rendered with ^^^.

Oh, there is just put a `push_primary_span` so I didn't know the context, the primary span that was added there is different than what I was thinking, I think we should either rename that or we should have some text letting others know what is primary span.

Reply:
Keep in mind that this detail is not exposed publicly. A user won't have to think about what kind of span is being added, as it's all in the same `mark`/`mark_all` API.